### PR TITLE
Restructure top-level menu

### PR DESCRIPTION
**Pull request by inteon**
Top-level menu proposal based on conversation I had with @wallrj.

https://github.com/cert-manager/website/pull/1309/commits/ae7d8bde74cf435fb11b8ebf00b919c1b61ea192 does the following:
1. move `/docs/usage/prometheus-metrics/` to `/docs/devops-tips/prometheus-metrics/`
2. move `/docs/tutorials/backup/` to `/docs/devops-tips/backup/`
3. move `/docs/tutorials/syncing-secrets-across-namespaces/` to `/docs/devops-tips/syncing-secrets-across-namespaces/`

4. move `/docs/projects/istio-csr/` to `/docs/usage/istio-csr/`
5. move `/docs/projects/csi-driver/` to `/docs/usage/csi-driver/`
6. move `/docs/projects/csi-driver-spiffe/` to `/docs/usage/csi-driver-spiffe/`
7. move `/docs/trust/trust-manager/` to `/docs/trust/trust-manager/`
8. move `/docs/trust/trust-manager/api-reference/` to `/docs/trust/trust-manager/api-reference/`
9. move `/docs/projects/approver-policy/` to `/docs/policy/approval/approver-policy/`
10. move `/docs/projects/approver-policy/api-reference/` to `/docs/policy/approval/approver-policy/api-reference/`

https://github.com/cert-manager/website/pull/1309/commits/887822653065d829768348f6fe78a2d14d7df49f contains the following new pages (explaining the different policies):
- `/docs/policy/`
- `/docs/policy/approval/`
- `/docs/policy/defaulting/`
- `/docs/policy/issuing/`

https://github.com/cert-manager/website/pull/1309/commits/5888c08268285aa03c981c19db36934ca41a3b07 contains the following new pages (explaining how to do trust properly):
- `/docs/trust/`

### CODE REVIEW

File: content/docs/concepts/certificate.md
Comment (wallrj) [inline content/docs/concepts/certificate.md:86]: I guess this moved to the trust intro....

File: content/docs/manifest.json
Comment (wallrj) [inline content/docs/manifest.json:43]: We'll remove this in a future PR.

File: content/docs/manifest.json
Comment (wallrj) [inline content/docs/manifest.json:57]: I thought we agreed not to make this change. I don't like it. I want Best practice to be more prominent than this.

File: content/docs/manifest.json
Comment (wallrj) [inline content/docs/manifest.json:61]: We're going to remove this page and add the verify instructions at the end of the kubectl apply page. In a future PR.

File: content/docs/policy/README.md
Comment (wallrj) [inline content/docs/policy/README.md:3]: This description doesn't make sense to me. 
I am a new small-scale user (not-multi-tenant) who wants to do the best thing by ensure that all the Certificates or csi-driver settings are using ECDSA keys and that private key rotation is set to always.
Someone sends me a link ending `.../policy/`  and it unfurls in Slack with the description "Rules for guiding certificate self-service".
I might assume that I'd been sent the wrong link.

**Review by wallrj — CHANGES_REQUESTED** [review]
I like the direction of this PR and I like the new policy pages and overview, but I think it needs further refinement and discussion.
And it goes much further than the PR title  "Restructure top-level menu".
And the PR description says what changes you've made, but it doesn't explain "Why?".

I suggest you create three or more PRs focussed on cleaning up the most pressing problems i.e.

1. We intend the top top-level menu items to be completed in a particular order, and we want to make that absolutely clear to the user.
2. The "Requesting Certificates" menu section no longer reflects its sub-menu items. Move prometheus metrics and backup to a devops section of the site.
3. The content in Concepts overlaps with the content in Requesting certificates and Configuring Issuers, and those overlaps should be reconciled.
4. The FAQ page contains redundant content which is already in the TLS terminology page.
5. Users  find it hard to understand what the cert-manager team means by "policy" so we should write it down.
6. Users find it hard to know what the cert-manager team consider to be secure defaults for Certificate and csi-driver.
7. There isn't an guide for devops engineers explaining all the things they may want to consider when deploying and configuring cert-manager.
8. There isn't a guide for large-scale users who want to make cert-manager secure in a multi-tenant cluster.

Each of these would be worthy of discussion and review.

**Review by wallrj — APPROVED** [review]
The team discussed these changes during the cert-manager standup on 2023-09-29 and agreed that it would be ok to merge this and make further improvements in later PRs.

/lgtm
/hold in case you want to make any further changes

**Comment by inteon** [comment]
During the standup we discussed this PR and concluded the best way forward is to merge this PR and make further improvements in follow-up PRs.

@wallrj

File: content/docs/policy/defaulting.md
Comment (wallrj) [inline content/docs/policy/defaulting.md:29]: Is that strictly true? 

You could default or override the "duration". For example: 
 * https://developer.venafi.com/tlsprotectcloud/docs/firefly-api-reference-clients#rest-methods

What about the "usages" field? That could be defaulted, if the PEM encoded CSR did not contain usages.
 * https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1.CertificateRequestSpec

File: content/docs/policy/defaulting.md
Comment (wallrj) [inline content/docs/policy/defaulting.md:34]: Suggest linking to a page explaining how ArgoCD can do this.

Comment (wallrj) [reply content/docs/policy/defaulting.md:34]: Kyverno could also be used to **override** field values of the Certificate, I think.

File: content/docs/policy/defaulting.md
Comment (wallrj) [inline content/docs/policy/defaulting.md:41]: Give an example. For example, the Let's Encrypt CA will ignore certain usages, I think...
 * https://letsencrypt.org/2015/11/09/why-90-days
 * https://community.letsencrypt.org/t/custom-extendedkeyusage-stripped-from-csr-while-generating-cert/17759/2

File: public/images/issuance-flow-policy.png
Comment (wallrj) [inline public/images/issuance-flow-policy.png:1]: It would be better to use SVG here or at least upload the source file, so that future maintainers can modify this diagram.

File: public/_redirects
Comment (wallrj) [inline public/_redirects:1]: Have you tested all these redirects?

File: public/images/issuance-flow-policy.png
Comment (inteon) [reply public/images/issuance-flow-policy.png:1]: I created this in powerpoint, so I don't have a svg and we probably don't want to upload the pptx file.
I should probably recreate this graphic in eg. draw.io.

File: public/_redirects
Comment (inteon) [reply public/_redirects:1]: I just tested them on the preview and turns out there was still an error.

File: content/docs/policy/defaulting.md
Comment (inteon) [reply content/docs/policy/defaulting.md:29]: I rephrased the sentence.

File: content/docs/policy/defaulting.md
Comment (inteon) [reply content/docs/policy/defaulting.md:34]: I actually wanted to mention Helm and customize instead of ArgoCD (any kind of templating tool).

Comment (inteon) [reply content/docs/policy/defaulting.md]: **Kyverno could also be used to override field values of the Certificate, I think.**

This is a good one, I did not consider this. I wonder if that means I should rewrite some of these docs, because now there are basically two places where you can have arbitrary mapping?

File: content/docs/manifest.json
Comment (inteon) [reply content/docs/manifest.json]: I reverted this change.

File: content/docs/policy/defaulting.md
Comment (inteon) [reply content/docs/policy/defaulting.md]: I will address this in a follow-up PR

**Pull request merged**
